File: Problems/python/boj_색종이.py
#https://www.acmicpc.net/problem/2563
#Implementation

import sys
input = sys.stdin.readline

n = int(input())
pl = [tuple(map(int, input().split())) for _ in range(n)]

m = max(map(max, pl)) + 10
mat = [[0]*m for _ in range(m)]

for c, r in pl:
    for i in range(r, r+10):
        for j in range(c, c+10):
            if mat[i][j]==0:
                mat[i][j] = 1
                
result = 0
for i in mat:
    result += sum(i)
print(result)

File: Problems/cpp/boj_색종이.cpp
//https://www.acmicpc.net/problem/2563
//Implementation

#include <iostream>

using namespace std;
 
int main(){
    ios::sync_with_stdio(0);
    cin.tie(0); cout.tie(0);

    int n, answer=0;
    cin >> n;

    bool paper[101][101] = {};

    int r, c;
    for (int k=0; k<n; k++){
        cin >> r >> c;
        for (int i=r; i<r+10; i++){
            for (int j=c; j<c+10; j++){
                if (paper[i][j]) continue;
                else{
                    paper[i][j] = true;
                    answer++;
                }
            }
        }
    }

    cout << answer;
}